script/Deploy.s.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.25;
+
+import {IEntryPoint} from "@eth-infinitism/account-abstraction/interfaces/IEntryPoint.sol";
+import {Script} from "forge-std/Script.sol";
+import {console2} from "forge-std/Test.sol";
+
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+
+import {AccountFactory} from "../src/account/AccountFactory.sol";
+import {UpgradeableModularAccount} from "../src/account/UpgradeableModularAccount.sol";
+import {SingleSignerValidation} from "../src/modules/validation/SingleSignerValidation.sol";
+
+contract DeployScript is Script {
+    IEntryPoint public entryPoint = IEntryPoint(payable(vm.envAddress("ENTRYPOINT")));
+
+    address public owner = vm.envAddress("OWNER");
+
+    address public accountImpl = vm.envOr("ACCOUNT_IMPL", address(0));
+    address public factory = vm.envOr("FACTORY", address(0));
+    address public singleSignerValidation = vm.envOr("SINGLE_SIGNER_VALIDATION", address(0));
+
+    bytes32 public accountImplSalt = bytes32(vm.envOr("ACCOUNT_IMPL_SALT", uint256(0)));
+    bytes32 public factorySalt = bytes32(vm.envOr("FACTORY_SALT", uint256(0)));
+    bytes32 public singleSignerValidationSalt = bytes32(vm.envOr("SINGLE_SIGNER_VALIDATION_SALT", uint256(0)));
+
+    uint256 public requiredStakeAmount = vm.envOr("STAKE_AMOUNT", uint256(0.1 ether));
+    uint256 public requiredUnstakeDelay = vm.envOr("UNSTAKE_DELAY", uint256(1 days));
+
+    function run() public {
+        console2.log("******** Deploying ERC-6900 Reference Implementation ********");
+        console2.log("Chain: ", block.chainid);
+        console2.log("EP: ", address(entryPoint));
+        console2.log("Factory owner: ", owner);
+
+        _deployAccountImpl(accountImplSalt, accountImpl);
+        _deploySingleSignerValidation(singleSignerValidationSalt, singleSignerValidation);
+        _deployAccountFactory(factorySalt, factory);
+        _addStakeForFactory(uint32(requiredUnstakeDelay), requiredStakeAmount);
+    }
+
+    function _deployAccountImpl(bytes32 salt, address expected) internal {
+        console2.log(string.concat("Deploying AccountImpl with salt: ", vm.toString(salt)));
+
+        address addr = Create2.computeAddress(
+            salt, keccak256(abi.encodePacked(type(UpgradeableModularAccount).creationCode, abi.encode(entryPoint)))
+        );
+        if (addr != expected) {
```

Comment:
Not sure if this is necessary in RI. It is gonna make test deployment annoying as you will need to compute all the addresses first (account impl, validation, etc.)

Reply:
It's just to prevent accidental drift between when it's calculated vs deployed, which can happen if using the wrong build profile, for example. For the first time pass, we can just run the script, and the error in `console.log` will tell us what address to put in.